tests/test_workflow.py

```diff
@@ -1,13 +1,52 @@
 """Tests for entire workflows."""
 from pathlib import Path
 
+import matplotlib.colors
 import matplotlib.testing.compare
 import numpy
 import xarray as xr
 from click.testing import CliRunner
+from matplotlib import pyplot as plt
 
 from wam2layers.cli import cli
 
+try:  # Use cartopy if it's available
+    import cartopy.feature as cfeature
+    from cartopy import crs
+except ImportError:
+    crs = None
+    cfeature = None
+
+
+def plot_difference(
+    expected_output: xr.DataArray, output: xr.DataArray, fpath: Path
+) -> None:
+    """Plot the difference between the expected output and actual output."""
+    if not fpath.parent.exists():
+        fpath.parent.mkdir()  # Make figures directory if it does not exist yet
+
+    if crs is not None:
+        subplot_kw = dict(projection=crs.PlateCarree())
+    else:
+        subplot_kw = None
+
+    fig, ax = plt.subplots(
+        figsize=(16, 10),
+        subplot_kw=subplot_kw,
+    )
```

Comment:
Although I like your solution for "maybe import cartopy", this gives me typing issues from pylance/pyright: 
```
"Argument of type "dict[str, PlateCarree] | None" cannot be assigned to parameter "subplot_kw" of type "dict[Unknown, Unknown]" in function "subplots""
```
You could set `subplot_kw` to `{}` instead of `None` and adapt the check on line 43 accordingly.

Reply:
Strange, the default value for `subplot_kw` is `None`. https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.subplots.html

Perhaps this is incorrectly typed in Microsoft's unofficial matplotlib stubs. Mypy doesn't detect a problem here.